src/DeployedEnvironment.ts

```diff
@@ -53,4 +54,19 @@ export class DeployedEnvironment {
             (contract: DeployedContract<Contract>) => addresses.indexOf(contract.deployedAddress.toLowerCase()) > -1
         );
     }
+
+    public getContractFromAddress(name: string, address: string): DeployedContract<Contract> {
```

Comment:
would probably be cleaner to make the return type optional (adding `| undefined`) rather than throwing.

you could then also remove all the `contractName` properties, which apparently was added everywhere only for the sake of this exception.

Reply:
I'm not sure about this. An explicit `MissingContractError` with the name and address can be more useful than an undefined value that breaks something later, without the context of what was not found where exactly.
Also the `name` is used as the key in `contracts` map (sometimes it's called `role`).
Maybe this comment refers to a previous code version?

Reply:
done. MissingContractError was replaced with plain Error